### PR TITLE
Make modals

### PR DESCRIPTION
**Pull request by valeriegc**

### CODE REVIEW

File: src/lib/Modal.svelte [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	import { onMount } from 'svelte';
+
+	let mounted = false;
+	onMount(() => {
+		mounted = true;
+	});
+	let dialog: HTMLDialogElement;
+	export let open: boolean;
+
+	$: if (open && mounted) {
```
Comment (AlexanderKolberg) [inline src/lib/Modal.svelte]: It looks like you are always wrapping this component in an if statement, cant this just be run onMont then?

File: src/routes/generator/+page.svelte [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+{/if}
+
 {#if $isMapModalOpen}
 	<MapModal />
 {/if}
 
 <div class="pageWrapper">
 	<Toolbar />
-	{#if !$files}
-		<PlaceHolder />
```
Comment (AlexanderKolberg) [inline src/routes/generator/+page.svelte:16]: If you are not using the placeholder anymore, should the file be deleted too?

File: src/routes/generator/StartModal.svelte [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		<div class="existingMaps">
+			<h3>View your existing maps</h3>
+		</div>
+		<div class="createNew">
+			<h3>Create from scratch</h3>
+			<div class="buttonContainer">
+				<input type="file" bind:files={$files} />
+			</div>
+			OR
+			<label style="margin-top:7%;background-color:transparent;font-weight:bold"
```
Comment (AlexanderKolberg) [inline src/routes/generator/StartModal.svelte]: Why does this have a background color of transparent? Isn't the label transparent by default?

File: src/routes/generator/StartModal.svelte [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		</div>
+	</div>
+</Modal>
+
+<style>
+	button {
+		background-color: transparent;
+	}
+	h3 {
+		background-color: transparent;
```
Comment (AlexanderKolberg) [inline src/routes/generator/StartModal.svelte]: Why is this needed?

File: src/routes/generator/stores.ts [hunk 5]
```diff
@@ -1,8 +1,10 @@
 import { writable } from 'svelte/store';
 
 export const isMapModalOpen = writable(false);
+export const startModalOpen = writable(false);
 export const files = writable<FileList | null>(null);
 export const mapObjects = writable<
 	{ x: number; y: number; link: string; width: number; heigth: number }[]
 >([]);
-export const userProvidedLink = writable('');
+export const userProvidedLink = writable('Please provide the link for your image');
```
Comment (AlexanderKolberg) [inline src/routes/generator/stores.ts]: Is this better suited to be in the placeholder text https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#placeholder

**Review by AlexanderKolberg — CHANGES_REQUESTED** [review]

File: src/routes/generator/+page.svelte (hunk 2 above)
Comment (valeriegc) [reply src/routes/generator/+page.svelte:16]: removed file.

File: src/routes/generator/StartModal.svelte (hunk 4 above)
Comment (valeriegc) [reply src/routes/generator/StartModal.svelte]: it was white by default.

File: src/routes/generator/StartModal.svelte (hunk 3 above)
Comment (valeriegc) [reply src/routes/generator/StartModal.svelte]: the label was white by default

File: src/routes/generator/StartModal.svelte (hunk 4 above)
Comment (valeriegc) [reply src/routes/generator/StartModal.svelte]: it was not transparet

File: src/routes/generator/StartModal.svelte (hunk 3 above)
Comment (valeriegc) [reply src/routes/generator/StartModal.svelte]: no it is not

File: src/routes/generator/stores.ts (hunk 5 above)
Comment (valeriegc) [reply src/routes/generator/stores.ts]: added placeholder

**Review by AlexanderKolberg — APPROVED** [review]
LGTM

**Review by AlexanderKolberg — APPROVED** [review]
LGTM
